### DISCUSSION TITLE
Very Complicated Issue...

### DISCUSSION THREAD

**Discussion by LucianAndries** [uBlockOrigin/uBlock-issues#1950] [Q&A] [answered]
Hello.

I'm having an issue with an unofficial steaming website, where getting rid of the ads breaks the embedded video iframe......
Can this be solved with a trick or so?

Let me explain. So these are the things I need blocked and unblocked. But it's complicated... For me anyway.

**Need Blocked:**
aniwatcher.com##div#Videoads

**Need Unblocked:**
aniwatcher.com#@#div#player
aniwatcher.com#@#div#player > div.spart
aniwatcher.com#@#div#player > div.spart > iframe.srcnopop[src^="//vidstreaming.io/streaming.php"]
aniwatcher.com#@#div#player > noscript

So, the things I need Unblocked, are the things that allow the embedding of a video on the website. So blocking the "ads injector" also makes the embedding iframe go bye-bye... 'Cause they made the friggin iframe a subcategory of the ads injector. I call it ads injector, 'cause I don't know what else to call it... Sneaky website!😡😭
I'm not that advanced with this stuff.

There should be a way to block the injector without blocking the embedding iframe. :(

Also, the website is as you guessed, again, aniwatcher.com... lol
I grew up in that community, so... I'm saddened they go to this heights for a couple of bucks. I know that's how they stay alive, but still, it's too intrusive.
The ads may have a close button, but when they show up again...................come on!😭😭

Here is a link for a video with aids: https://aniwatcher.com/bleach-episode-290

**Screenshots:**

![in video ad](https://user-images.githubusercontent.com/59364663/151024588-30ba2c3f-d003-44c4-9f8e-73ad00c0ebfe.jpg)
.
![Logger](https://user-images.githubusercontent.com/59364663/151024645-6216c2f2-eda6-4752-b873-bfccefd79423.jpg)
.
![Inspect](https://user-images.githubusercontent.com/59364663/151024675-f5867319-a0be-4311-a98d-410de1344927.jpg)
.
_______________________________________________________________________________________________________________

**PS:** Is there a way to block/bypass a friggin anti-adblock warning? Using uBlock's tools of course. Or at least can someone recommend me a very good filter/plugin that kills these things?
Because not even setting the website to Green or turning off uBlock make them stop anymore... I have to disable the entire extension for that to stop. 😭😭😭
It's for this website(video): https://fembed-hd.com/v/kyjk1s3gx7133-7

**EDIT:**

Pffffffffffffffff, nevermind. I found a fix, for this website at least........ So if you Cosmetic Filtering and/or Remote Fonts, you won't get the anti-adblock warning anymore.
Can anyone confirm this fix on their favourite streaming hosts?
But, doing this brings back the already blocked ads...

**Comment by mapx-** [comment, marked as answer]
- You have to test the site using **only default settings => reset uBO to the default settings**
- provide exact page where you still see ads

> Need Unblocked

Explain why ? provide exact page

**Reply by LucianAndries** [reply]
Hey.

I thought I did, sorry... Here is a page: `https://aniwatcher.com/bleach-episode-290`
Once you go fullscreen, the ads show up. It is a IN VIDEO ad.

Explain why, what? Why I need that unblocked? Guess I was very vague with this one. Sorry again.
The things I need unblocked, are the things that allow the embedding of a video on the website. So blocking the ads injector also makes the embedding iframe go bye-bye... 'Cause they made the friggin iframe a subcategory of the ads injector.

There should be a way to block the injector without blocking the embedding iframe. :(

Also: Testing it with default settings... What makes you think I'm not? Please explain your reasoning.
Because the things you see red/blocked on the Logger, are just the empty spaces left on the website, when the normal ads were blocked. That's all. I don't like empty spaces, they make you scroll for years!

**Comment by uBlock-user** [comment]
> What makes you think I'm not? Please explain your reasoning.

https://github.com/uBlockOrigin/uBlock-issues/discussions/1906

**Reply by uBlock-user** [reply]
Don't use Dynamic Filtering, it's not for you.

**Reply by LucianAndries** [reply]
What is dynamic filtering?

**Comment by LucianAndries** [comment]
Pfffffff, Labels: unable to reproduce...........................🤦‍♂️

I don't understand WHAT is so hard to do?
-You load the link I gave: `https://aniwatcher.com/bleach-episode-290`
-You click play video, as is.
-If the Ad doesn't show up, then click full screen. It will show up in 2 seconds.

I'm using the default settings.

**Reply by mapx-** [reply]
> default settings

Did you really use this button bellow (last one) ? 
Then go into "filter lists" tab, click "purge all caches", click "update now".
Then don't add any other custom filter or other list. 
Don't use dynamic filters. 
Just test again.

![image](https://user-images.githubusercontent.com/10303732/151201397-94a54cd1-8c41-478d-9d24-8d7229223d82.png)

**Reply by LucianAndries** [reply]
Yeah, not working. I had to reinstall it.

Seems to be working now, and seems to be blocking a lot of stuff I was manually blocking. The filters have come this far? Well, It's 1 website so I won't celebrate yet.
And the injected crap into video seems to be gone, along with a few popunders too...

Now, the only custom filters I had were for anti: porn, malware, adware and crypto. Can I add them back, or did the default filters get to a real good state?

Ok, are "dynamic filters" the ones you block yourself? Why isn't anyone make it clear for me? :(

One issue though. I want the old theme back, because the drop-menu is too big. I wrote "classic" in the advanced settings, but it doesn't want to change anymore... What changed?

**Comment by LucianAndries** [comment]
Also, the new theme is bugged.
It opens the Logger a new window, then a new tab, then a new window, etc, etc.
And when it opens in a new window, the page is black. You need to move the window to make everything show up.

PS: I would very much appreciate a dark mode for uBlock. I bet all of us would! Has it been discussed yet? :(

**Reply by mapx-** [reply]
- keep SHIFT pressed when you open the logger.
- don't mess again with other lists and don't touch the dynamic settings (right side stuff). The lists are ALWAYS in good state.
- any other info => read the wiki: https://github.com/gorhill/uBlock/wiki

**Reply by LucianAndries** [reply]
Keep shift pressed...... Yeah, WHY??? It should open in a new tab, always, by itself! And no, holding shift does nothing. It is a bug, or a dumb design. No offence to the creator.

As for dynamic filtering, it's the one on the left, not right. I read the wiki... Instead of setting noop to websites, I was fully allowing them, thus ignoring all other filters. Meaning the ads were allowed as well. I misunderstood how those worked..... My bad.

Still, how the hell do I change the theme to the classic one? :((((((((

**Reply by mapx-** [reply]
Obviously - **only at the beginning** ! - **SHIFT + click the logger icon**.

**Reply by uBlock-user** [reply]
>Yeah, WHY??

Explained in the wiki - https://github.com/gorhill/uBlock/wiki/The-logger#tip Read the wiki, these are FAQs that are answered in there.

> how the hell do I change the theme to the classic one?

Explained in the wiki for Advanced settings - https://github.com/gorhill/uBlock/wiki/Advanced-settings
